install/kubernetes/cilium/charts/agent/templates/servicemonitor.yaml

```diff
@@ -1,4 +1,4 @@
-{{- if and .Values.global.prometheus.enabled (.Values.global.prometheus.serviceMonitor.enabled) }}
+{{- if and .Values.global.prometheus.enabled .Values.global.prometheus.serviceMonitor.enabled (.Capabilities.APIVersions.Has "monitoring.coreos.com/v1") }}
```

Comment:
Understand that this approach is quite common in multiple public helm charts, I still have one small concern about user experience. Basically,  user might install cilium chart (with skip crd flag) and assume that ServiceMonitor CRD is installed, but it's silently ignored. 

I am ok with either cases though.